server/src/main/java/org/opensearch/search/profile/ProfileResult.java

```diff
@@ -168,9 +275,22 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         builder.field(DESCRIPTION.getPreferredName(), description);
         if (builder.humanReadable()) {
             builder.field(NODE_TIME.getPreferredName(), new TimeValue(getTime(), TimeUnit.NANOSECONDS).toString());
+            if (concurrent) {
+                builder.field(MAX_SLICE_NODE_TIME.getPreferredName(), new TimeValue(getMaxSliceTime(), TimeUnit.NANOSECONDS).toString());
+                builder.field(MIN_SLICE_NODE_TIME.getPreferredName(), new TimeValue(getMinSliceTime(), TimeUnit.NANOSECONDS).toString());
+                builder.field(AVG_SLICE_NODE_TIME.getPreferredName(), new TimeValue(getAvgSliceTime(), TimeUnit.NANOSECONDS).toString());
+            }
         }
         builder.field(NODE_TIME_RAW.getPreferredName(), getTime());
-        builder.field(BREAKDOWN.getPreferredName(), breakdown);
+        Map<String, Long> modifiedBreakdown = new LinkedHashMap<>(breakdown);
+        if (concurrent) {
+            builder.field(MAX_SLICE_NODE_TIME_RAW.getPreferredName(), getMaxSliceTime());
+            builder.field(MIN_SLICE_NODE_TIME_RAW.getPreferredName(), getMinSliceTime());
+            builder.field(AVG_SLICE_NODE_TIME_RAW.getPreferredName(), getAvgSliceTime());
+        } else {
+            removeStartTimeFields(modifiedBreakdown);
```

Comment:
I think we should not modifying anything here, just copy + filter

Reply:
@ticheng-aws my apologies, the `modifiedBreakdown` is indeed a local copy, I meant something like this:

```
Map<String, Long> modifiedBreakdown = createBreakownView(concurrent, breakdown);
```

So instead copying and removing entries, we would just get the right map by filtering out the non-relevant ones

Reply:
Got it. We can do that as well. Please help to review it again once you get a chance. Thank you.